File: course_android/LittleDemo/mobileinfo/src/main/java/better/learn/mobileinfo/MainApp.java
package better.learn.mobileinfo;

import android.app.Application;

/**
 * Created by better on 2016/12/28.
 */

public class MainApp extends Application {
    static MainApp mMainApp;
    @Override
    public void onCreate() {
        super.onCreate();
        mMainApp=this;
    }

    static Application getInstance(){
        return mMainApp;
    }
}

File: course_android/HelloKotlin/app/src/main/java/better/hellokotlin/MainApp.kt
package better.hellokotlin

import android.app.Application
import kotlin.properties.Delegates

/**
 * Created by better on 2017/7/28 15:55.
 */
class MainApp : Application() {
    companion object {
        var instance: MainApp by Delegates.notNull<MainApp>()
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

//    fun getInstance(): MainApp = instance
}